### PR TITLE
Implement process.name

### LINKED ISSUE #58: Implement process.name
**Is your feature request related to a problem? Please describe.**
Eldritch does not have a simple way to get the name of a process from its PID.

**Describe the solution you'd like**
Implement a process.name function.
- Should take an int PID and return a string that is the name of that process.

### PR DESCRIPTION
**Pull request by adm1nPanda**
modified:   docs/_docs/user-guide/eldritch.md
	modified:   implants/lib/eldritch/src/process/name_impl.rs

#### What type of PR is this?
/kind feature
/kind eldritch-function

#### What this PR does / why we need it:
Implement process.name as per #58

#### Which issue(s) this PR fixes:

Fixes #58

### CODE REVIEW

File: implants/lib/eldritch/src/process/name_impl.rs [hunk 1]
```diff
@@ -1,5 +1,67 @@
...
+                cfg!(target_os = "ios") || 
+                cfg!(target_os = "macos") || 
+                cfg!(target_os = "android") || 
+                cfg!(target_os = "freebsd") || 
+                cfg!(target_os = "openbsd") ||
+                cfg!(target_os = "netbsd") {
+            commandstring = "sleep";
+        } else if cfg!(target_os = "windows") {
+            commandstring = "timeout";
+        }
```
Comment (hulto) [inline implants/lib/eldritch/src/process/name_impl.rs:41]: Could you add an else here to make it clear the test will fail on non-supported OSes.
```rust
else {
    return anyhow::anyhow!(Err("OS Not supported please re run on Linux, Windows, or MacOS");
}
```

**Review by hulto — CHANGES_REQUESTED** [review]
Hey adm1nPanda,
Thank you for submitting a PR!
We're excited to have you contribute I've left a few questions and suggestions. Really great job on your first PR 🙂

**Review by hulto — APPROVED** [review]
Looks great! 🚀

**Comment by adm1nPanda** [comment]
windows workflow failing in actions but passes in my lab. made some changes which i hope will clear this time

File: implants/lib/eldritch/src/process/name_impl.rs [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
+use sysinfo::{Pid, PidExt, ProcessExt, System, SystemExt};
 
-pub fn name(_pid: i32) -> Result<String> {
-    unimplemented!("Method unimplemented")
-}
\ No newline at end of file
+pub fn name(pid: i32) -> Result<String> {
+    if !System::IS_SUPPORTED {
+        return Err(anyhow::anyhow!("This OS isn't supported for process functions.
+         Pleases see sysinfo docs for a full list of supported systems.
```
Comment (KCarretto) [inline implants/lib/eldritch/src/process/name_impl.rs:7]: Nit: Please see*

**Comment by hulto** [comment]
Looks good! Go ahead and sqaush and merge.

**Review by hulto — APPROVED** [review]

**Pull request merged**